### PR TITLE
New Port: Biomass Growth

### PR DESCRIPTION
**Pull request by Ganimoth**
# New Port for Biomass Growth

## Game Information
- **Title**: Biomass Growth
- **URL**: https://brainos.itch.io/biomassgrowth

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

File: ports/biomassgrowth/BIOMASS Growth.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+source $controlfolder/control.txt
+# device_info.txt will be included by default
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+GAMEDIR="/$directory/ports/biomassgrowth"
+
+export PORT_32BIT="Y"
```
Comment (Cebion) [inline ports/biomassgrowth/BIOMASS Growth.sh:23]: Has to be above mod_ lines

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/biomassgrowth/BIOMASS Growth.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+# Log the execution of the script into log.txt
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+cd $GAMEDIR
+
+# Extract the game files
+if [ -f "$GAMEDIR/gamedata/Biomass Growth 20211213.exe" ]; then
+    
+	cd "$GAMEDIR" || exit
```
Comment (cdeletre) [inline ports/biomassgrowth/BIOMASS Growth.sh:36]: It already changed directory line 31 :) Sure you need it ?

File: ports/biomassgrowth/BIOMASS Growth.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		
+	# Use 7zip to extract the Biomass Growth 20211213.exe file to the destination directory
+    "$GAMEDIR/patch/7zzs" -aos x "$GAMEDIR/gamedata/Biomass Growth 20211213.exe" -o"$GAMEDIR/gamedata" & pid=$!
+
+    # Wait for the extraction process to complete
+    wait $pid
+
+    # Check if the Biomass Growth 20211213.exe file exists
+    if [ -f "$GAMEDIR/gamedata/Biomass Growth 20211213.exe" ]; then
```
Comment (cdeletre) [inline ports/biomassgrowth/BIOMASS Growth.sh:45]: It was already tested in the `if` statement line 34. Do we need to test again ? I haven't looked at the content of `Biomass Growth 20211213.exe` so maybe I am missing something :)

File: ports/biomassgrowth/BIOMASS Growth.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# Extract the game files
+if [ -f "$GAMEDIR/gamedata/Biomass Growth 20211213.exe" ]; then
+    
+	cd "$GAMEDIR" || exit
+		
+	# Use 7zip to extract the Biomass Growth 20211213.exe file to the destination directory
+    "$GAMEDIR/patch/7zzs" -aos x "$GAMEDIR/gamedata/Biomass Growth 20211213.exe" -o"$GAMEDIR/gamedata" & pid=$!
+
+    # Wait for the extraction process to complete
+    wait $pid
```
Comment (cdeletre) [inline ports/biomassgrowth/BIOMASS Growth.sh:42]: Do we really need this `pid` and `wait` mechanism ? Wouldn't just this line work ?

```
"$GAMEDIR/patch/7zzs" -aos x "$GAMEDIR/gamedata/Biomass Growth 20211213.exe" -o"$GAMEDIR/gamedata"
```

File: ports/biomassgrowth/biomassgrowth/game.apk
Comment (cdeletre) [inline ports/biomassgrowth/biomassgrowth/game.apk:1]: The apk needs some clean-up. Just keep the `lib/armeabi` and `lib/armeabi-v7a` folders :)
(I didn't know a mips build of libyoyo existed btw)

```
unzip -l game.apk
Archive:  game.apk
  Length      Date    Time    Name
---------  ---------- -----   ----
     5472  00-00-1980 00:00   AndroidManifest.xml
     1361  00-00-1980 00:00   META-INF/CERT.RSA
     1877  00-00-1980 00:00   META-INF/CERT.SF
     1834  00-00-1980 00:00   META-INF/MANIFEST.MF
  1406608  00-00-1980 00:00   classes.dex
  4772484  00-00-1980 00:00   lib/armeabi-v7a/libyoyo.so
  4969108  00-00-1980 00:00   lib/armeabi/libyoyo.so
  6850476  00-00-1980 00:00   lib/mips/libyoyo.so
  6055484  00-00-1980 00:00   lib/x86/libyoyo.so
      244  00-00-1980 00:00   res/drawable-hdpi-v4/icon.png
      217  00-00-1980 00:00   res/drawable-ldpi-v4/icon.png
      228  00-00-1980 00:00   res/drawable-mdpi-v4/icon.png
      270  00-00-1980 00:00   res/drawable-xhdpi-v4/icon.png
      323  00-00-1980 00:00   res/drawable-xxhdpi-v4/icon.png
      392  00-00-1980 00:00   res/drawable-xxxhdpi-v4/icon.png
      270  00-00-1980 00:00   res/drawable/icon.png
     5072  00-00-1980 00:00   res/layout/download.xml
      808  00-00-1980 00:00   res/layout/main.xml
     1528  00-00-1980 00:00   res/layout/userpasslayout.xml
     1040  00-00-1980 00:00   res/xml/preferences.xml
     5448  00-00-1980 00:00   resources.arsc
---------                     -------
 24080544                     21 files
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/biomassgrowth/BIOMASS Growth.sh (hunk 2 above)
Comment (Ganimoth) [reply ports/biomassgrowth/BIOMASS Growth.sh:36]: Yeah, thats a duplicate

File: ports/biomassgrowth/biomassgrowth/game.apk
Comment (Ganimoth) [reply ports/biomassgrowth/biomassgrowth/game.apk:1]: I see there is a cleaned up wrapper in the fraxinuses repo, seems leaner, so I will try that one.

File: ports/biomassgrowth/BIOMASS Growth.sh (hunk 4 above)
Comment (Ganimoth) [reply ports/biomassgrowth/BIOMASS Growth.sh:42]: I tried to run a couple times without and seems like I dont need it there.

**Review by Cebion — APPROVED** [review]

**Review by cdeletre — APPROVED** [review]

File: ports/biomassgrowth/biomassgrowth/game.apk
Comment (cdeletre) [reply ports/biomassgrowth/biomassgrowth/game.apk:1]: All good :)

**Comment by Ganimoth** [comment]
I have accidentally closed it too soon

**Pull request merged**